src/containers/buttons/DeleteTestButton.js

```diff
@@ -0,0 +1,48 @@
+import React, {Component} from 'react';
+import {connect} from "react-redux";
+import {bindActionCreators} from 'redux';
+import {deleteTestFromProject, clearTests, hideUrlInput, clearUrlsList, hideTestNameInput} from "../../actions"
+
+class DeleteTest extends Component {
+
+  deleteTest() {
+   if(this.props.test.length > 0 && this.props.testName.name) {
+    if(this.checkIfTestAlreadyExists(this.props.testName)) {
+      this.props.deleteTestFromProject(this.props.testName);
+      this.props.clearTests();
+      this.props.hideUrlInput();
+      this.props.clearUrlsList();
+      this.props.hideTestNameInput(true);
+    }
+   }
+  }
+
+  checkIfTestAlreadyExists(testName) {
+    let exists = false;
+    document.querySelectorAll(".tests-container>.test-item>.test-name").forEach((test) => {
```

Comment:
I'd rather see that done programatically by setting react state than querying dom directly, if not you can make regular array from that and use find to achieve similar functionality but with less loop executions

Reply:
Done.